share/steel/examples/pulse/lib/Pulse.Lib.Par.Pledge.fst

```diff
@@ -199,8 +199,9 @@ let inv_p' (os0 : inames) (f v1 v2 : vprop) (r1 r2 : GR.ref bool) (b1 b2 : bool)
 let inv_p (os0 : inames) (f v1 v2 : vprop) (r1 r2 : GR.ref bool) : vprop =
   exists* b1 b2. inv_p' os0 f v1 v2 r1 r2 b1 b2
 
+[@@expect_failure]
```

Comment:
This fails, even when I annotate it as `atomic` instead of `ghost`. 
But, it ought to work.
The problem here is that the body of elim_l looks like 
```
let x = something_with_ghost_effect in
...
return_stt_atomic v
```
where `x` has an informative type.
Since the return_stt_atomic is in Unobservable, the composition with a ghost computation returning an informative value does not work. 
I have work in progress in another branch to add another "Neutral" effect which composes with both ghost and atomic, but that's not ready yet.
With a neutral effect, I think I can restore this code to work in atomic. But, that's still not enough for the pledge library, since it will not be ghost.

Reply:
Actually, I realized I can make it work by factoring the ghost code into its own annotated block.
If we had a neutral effect. then we wouldn't need this kind of factoring of ghost code.